tls/s2n_prf.c

```diff
@@ -205,14 +205,14 @@ static int s2n_evp_hmac_p_hash_free(struct s2n_prf_working_space *ws)
     return 0;
 }
 
-static const struct s2n_p_hash_hmac s2n_evp_hmac = {
-    .alloc = &s2n_evp_hmac_p_hash_new,
-    .init = &s2n_evp_hmac_p_hash_init,
-    .update = &s2n_evp_hmac_p_hash_update,
-    .final = &s2n_evp_hmac_p_hash_digest,
-    .reset = &s2n_evp_hmac_p_hash_reset,
-    .cleanup = &s2n_evp_hmac_p_hash_cleanup,
-    .free = &s2n_evp_hmac_p_hash_free,
+static const struct s2n_p_hash_hmac s2n_evp_pkey_hmac = {
+    .alloc = &s2n_evp_pkey_hmac_alloc,
+    .init = &s2n_evp_pkey_hmac_init,
+    .update = &s2n_evp_pkey_hmac_update,
+    .final = &s2n_evp_pkey_hmac_final,
+    .reset = &s2n_evp_pkey_hmac_reset,
+    .cleanup = &s2n_evp_pkey_hmac_cleanup,
+    .free = &s2n_evp_pkey_hmac_free,
```

Comment:
I looked into this more, it looks like this was named because it is used in the TLS P_hash function, but it is actually just a HMAC wrapper. The P_hash construction is what s2n implements in `s2n_p_hash` which uses `s2n_p_hash_hmac`. See https://datatracker.ietf.org/doc/html/rfc2246#section-5